R/fit.R

```diff
@@ -415,12 +418,21 @@ mmrm <- function(formula,
     drop_visit_levels = control$drop_visit_levels,
     allow_na_response = FALSE
   )
-  fit <- fit_single_optimizer(
-    tmb_data = tmb_data,
-    formula_parts = formula_parts,
-    control = control
-  )
-
+  fit <- list()
+  while ((length(fit) == 0) && length(control$optimizers) > 0) {
+    fit <- fit_single_optimizer(
+      tmb_data = tmb_data,
+      formula_parts = formula_parts,
+      control = control
+    )
+    if (length(fit) == 0) {
+      warning(paste0(
+        "Divergence with optimizer ", names(control$optimizers[1L]), " due to problems: ",
+        toString(attr(fit, "divergence"))
+      ))
+    }
+    control$optimizers <- control$optimizers[-1]
```

Comment:
if all optimizer fails, `fit` is length 0 and there are no successful optimizers. then is there need to call `refit_multiple_optimizers` again?

Reply:
this first while part is only finding the first optimizer returning a non-empty fit. Exactly, if all fail then below we will get the error to try multiple optimizers and refit_multiple_optimizers is not called. also have a look at the unit tests I added for these edge cases